### PR TITLE
Updated census household & communal establishment schemas to match slides

### LINKED ISSUE #665: Census_English: Is displaying Welsh only question for English region.
### Expected behaviour
Must not display question 16. Can you understand, speak, read or write Welsh? when the survey is a English region survey

### Actual behaviour
question 16. Can you understand, speak, read or write Welsh? is getting displayed for an english region survey

### Steps to reproduce the behaviour
Launch Census_English region survey
Navigate to question 14a in Individual and click [save and continue]

### LINKED ISSUE #680: Census: When 'No' Selected or No option selected for Q:- Have you ever worked? Next question is asked
### Expected behaviour ###
When 'No' Selected or No option selected for Q:- Have you ever worked? then that must be the last question for section and must navigate to Completion Interstitial page

### Actual behaviour ###

Navigating to next question asking details about Job which is not correct

### Steps to reproduce the behaviour ###
Launch Census survey
Navigate to Have you ever worked? 
Select No or do not select any option 
Click Save and Continue

### Browser ###

All

### Screenshot ###

### LINKED ISSUE #682: Census_Communal: Validation error not displayed when No option selected on Q1 in Usual Residents section
### Expected behaviour ###

When no option is selected on Question:1 Are there any usual residents living at this establishment?
must display a alert - Please provide an answer to continue’

### Actual behaviour ###

Navigates to next question when no option is selected 

### Steps to reproduce the behaviour ###
Launch Census_communal survey
Navigate to second question
Do not select any option click save and continue

### Browser ###

All

### Screenshot ###

### LINKED ISSUE #684: Census: Navigation from Question '28 Last week were you:' was incorrect when more than 2 options out of the top 6 options are selected
### Expected behaviour

Must skip the next 5 questions

### Actual behaviour
No Questions are skipped and application simply navigates to next question which is inccorect.
**Note:** Next five questions are skipped when only one option from the top 6 option(s) are selected

### Steps to reproduce the behaviour
Launch Census_household survey
Navigate to Question 26. Last week were you:
And select any two or more options from the top 6 options and click save and continue

#### Browser
All

### LINKED ISSUE #685: Census: When 'No' is selected for first 2 questions and 0 visitors then the navigation is not correct
### Expected behaviour ###

Must directly take you to confirm Number of visitors page?

### Actual behaviour ###

Taking the user to Question.5 Does your household own or rent this accommodation

### Steps to reproduce the behaviour ###
Launch a Census Survey from RH
Answer 'No' for Question 1 Does anyone live here?
Answer 'No' for Question 1a is there anyone else who live here?
Answer '0' for Number of visitors 
Answer Household questions till question 4 and click save and continue.

Technical information

### Browser ###

All

Browser

Operating System

Screenshot

### LINKED ISSUE #698: Census_Household_Wales: English version of Q: How would you describe your national identity is displayed
### Expected behaviour ###
Must display Wales version of the Question

### Actual behaviour
Displayed English version of the Question and all the *** following questions are also english versions which is not correct ***

**Note:** The previous Question What is your country of birth is displayed in the correct version (Wales)
### Steps to reproduce the behaviour
Launch census_household_Wales survey
Navigate to What is your country of birth select country and navigate to 
How would you describe your national identity ?

### Screenshot

screen shot 2016-12-12 at 12 51 43

### LINKED ISSUE #700: Census Completion Preferences question incorrectly formatted
### Expected behaviour

Formatting should be as per the google slides with the question appearing separated from the section description.

### Actual behaviour

Question appears squashed up against section description.

### Steps to reproduce the behaviour

Select the census_communal schema. Progress to completion preferences page.

### LINKED ISSUE #701: Cenus_Communal: Error message not correct on Establishment 2nd Question
### Expected behaviour ###
Must display an error message 'Please provide an answer to continue’

### Actual behaviour
Displays incorrect error message 'This field is mandatory'

### Steps to reproduce the behaviour
Launch census_communal survey
Navigate to navigate to 2nd question in Establishment section 
Click [Save and Continue] without entering any value 

### Screenshot

screen shot 2016-12-12 at 15 38 09

### PR DESCRIPTION
**Pull request by jonnyshaw89**
### What is the context of this PR?
Updating the census schemas to match the defined content in the slides

Fixes #665
Fixes #680
Fixes #682 
Fixes #684
Fixes #685
Fixes #698
Fixes #700
Fixes #701

### How to review 
Check that content match slides

### CODE REVIEW

File: app/templates/partials/questions/repeatinganswer.html [hunk 1]
```diff
@@ -1,9 +1,36 @@
...
+  <h2 class="question__title neptune">
+    {%- if question_number -%}
+      <span>{{question_number}}. </span>
+    {% endif %}
+    {{question_title}}
+  </h2>
 
   {{question_description|safe}}
 
+  {%- if question.schema_item.guidance -%}
```
Comment (iwootten) [inline app/templates/partials/questions/repeatinganswer.html]: I don't think this is necessary...there's updates to display guidance in my summary branch.

**Review by iwootten — CHANGES_REQUESTED** [review]

File: app/templates/partials/questions/repeatinganswer.html (hunk 1 above)
Comment (benjefferies) [reply app/templates/partials/questions/repeatinganswer.html]: It will get sorted out in one of the merges anyway?

Comment (jonnyshaw89) [reply app/templates/partials/questions/repeatinganswer.html]: resolved this conflict

File: app/templates/partials/questions/relationship.html [hunk 2]
```diff
@@ -2,7 +2,13 @@
 
 <div class="relationship">
 
-    <h2 class="neptune">{{question_title}}</h2> {{question_description|safe}}
+    <h2 class="question__title neptune">
+      {%- if question_number -%}
+        <span>{{question_number}}. </span>
+      {% endif %}
+      {{question_title}}
```
Comment (iwootten) [inline app/templates/partials/questions/relationship.html]: Should we be piping the question title through safe as we do elsewhere to?

**Comment by collisdigital** [comment]
@jonnyshaw89 I've pushed some changes up based on a review today. I had to add another routing rule that goes to `household-member-completed` block (which doesn't exist yet).

Most importantly I had to remove the group_id check from navigator.py to allow you to navigate to a block in another group (i.e. the confirmation screen or the start of the visitors section). @iwootten may want to comment on that. Discussing with @ajmaddaford we probably need a way to jump to a group_id which takes you to the first block in the group (thus avoiding the above change). He's going to look at that as part of the Navigation branch but probably merge that first then work on a refractor as part of Ian's location refactor work (which also relates to #693)

In short, I think careful planning tomorrow of what to do is needed!

File: app/data/census_household.json [hunk 3]
```diff
@@ -3941,7 +3920,7 @@
                         {
                             "id": "visitor-sex-section",
                             "number": "2",
-                            "title": "Visitor {{ [answers.visitor_first_name, answers.visitor_last_name] | format_household_name }}",
+                            "title": "Visitors",
```
Comment (jonnyshaw89) [inline app/data/census_household.json]: @collisdigital have we changed our minds on having the visitors name?

File: app/data/census_household.json [hunk 4]
```diff
@@ -1163,6 +1194,13 @@
                                 }
                             ]
                         }
+                    ],
+                    "routing_rules": [
+                        {
+                            "goto": {
+                                "id": "household-member-completed"
```
Comment (jonnyshaw89) [inline app/data/census_household.json:1211]: where is this page coming from?

Comment (iwootten) [reply app/data/census_household.json:1211]: Yeah, don't see the point of adding a routing rule that is going to break, especially if we're modifying navigator in order to make it jump outside of a group.

**Review by iwootten — CHANGES_REQUESTED** [review]

File: app/questionnaire/navigator.py [hunk 5]
```diff
@@ -39,8 +39,9 @@ def is_interstitial_block(cls, block_id):
     @classmethod
     def _block_index_for_location(cls, blocks, location):
         if not cls.is_interstitial_block(location['block_id']):
-            return next(index for (index, b) in enumerate(blocks) if b["block"]["id"] == location['block_id'] and
-                        b["group_id"] == location['group_id'] and b['group_instance'] == location['group_instance'])
+            return next(index for (index, b) in enumerate(blocks) if
```
Comment (iwootten) [inline app/questionnaire/navigator.py]: This seems dangerous to me. A lot of the repeat logic is based around being within the same group/instance. It's not clear what impact removing the group id check has. Seems like routing rules should be able to specify both group/block_id.

File: app/data/census_household.json [hunk 6]
```diff
@@ -2893,16 +3039,16 @@
                     "sections": [
                         {
                             "id": "disability-section",
```
Comment (benjefferies) [inline app/data/census_household.json:3328]: Should there be routing on this to skip to 27. Thinking of the last 12 months... if the person is under 16?

**Comment by iwootten** [comment]
The formatting on these Communal Establishment pages looks pretty bad:
screen shot 2016-12-12 at 12 29 58
screen shot 2016-12-12 at 12 32 11

File: app/data/census_household.json (hunk 4 above)
Comment (jonnyshaw89) [reply app/data/census_household.json:1211]: sorted

File: app/data/census_household.json (hunk 6 above)
Comment (jonnyshaw89) [reply app/data/census_household.json:3328]: done

**Comment by iwootten** [comment]
There's issues with this household page. The section number and the question number is shown, where there isn't a section number in the slides.

screen shot 2016-12-12 at 12 50 46

**Comment by iwootten** [comment]
There's an issue on the completed page, where the title in the slides is different to the schema
screen shot 2016-12-12 at 13 06 33
screen shot 2016-12-12 at 13 06 41

File: app/data/census_household.json [hunk 7]
```diff
@@ -430,20 +433,20 @@
...
+                                    "description": "In the next section we are going to ask you about the household and accommodation you live in.",
                                     "answers": [],
                                     "id": "who-lives-here-completed-question",
-                                    "title": "Next section:",
+                                    "title": "",
                                     "type": "General"
                                 }
                             ],
-                            "title": "You have completed"
+                            "title": "You have successfully completed the 'Who lives here?' section"
```
Comment (benjefferies) [inline app/data/census_household.json:463]: Wording in slides are `You have successfully completed this section`

Comment (jonnyshaw89) [reply app/data/census_household.json:463]: It is, but this has changed since.

File: app/data/census_household.json [hunk 8]
```diff
@@ -2430,13 +2542,13 @@
...
                         {
                             "id": "language-section",
-                            "number": "17",
-                            "title": "{{ [answers.first_name[group_instance], answers.last_name[group_instance]] | format_household_name }}",
+                            "number": "19",
+                            "title": "{{[answers.first_name[group_instance], answers.last_name[group_instance]] | format_household_name }}",
                             "questions": [
                                 {
                                     "id": "language-question",
                                     "title": "What is your main language",
```
Comment (benjefferies) [inline app/data/census_household.json]: Missing question mark

File: app/data/census_household.json [hunk 9]
```diff
@@ -2471,7 +2582,7 @@
                                 {
                                     "id": "language-welsh-question",
                                     "title": "What is your main language",
```
Comment (benjefferies) [inline app/data/census_household.json]: Missing question mark

File: app/data/census_household.json [hunk 10]
```diff
@@ -2867,24 +2970,63 @@
...
+                            "number": "23",
+                            "questions": [
+                                {
+                                    "id": "other-passports-question",
+                                    "title": "Please specify other passports held",
+                                    "number": "23a",
+                                    "type": "General",
+                                    "answers": [
+                                        {
+                                            "id": "other-passports-answer",
```
Comment (benjefferies) [inline app/data/census_household.json:3312]: Needs `Passports held` label

File: app/data/census_household.json [hunk 11]
```diff
@@ -3128,13 +3284,13 @@
...
+                            "title": "{{[answers.first_name[group_instance], answers.last_name[group_instance]] | format_household_name }}",
                             "questions": [
                                 {
                                     "id": "volunteering-question",
                                     "title": "Thinking of the last 12 months, have you taken part in any volunteering for any groups, clubs or organisations?",
-                                    "number": "25",
+                                    "number": "27",
                                     "guidance": [
                                         {
                                             "description": "Exclude any Court ordered activities"
```
Comment (benjefferies) [inline app/data/census_household.json]: Exclude should be in bold

File: app/data/census_household.json [hunk 12]
```diff
@@ -3178,13 +3333,13 @@
...
+                            "title": "{{[answers.first_name[group_instance], answers.last_name[group_instance]] | format_household_name }}",
                             "questions": [
                                 {
                                     "id": "employment-type-question",
                                     "title": "Last week were you:",
-                                    "number": "26",
+                                    "number": "28",
                                     "description": "Select all that apply",
                                     "guidance": [
                                         {
```
Comment (benjefferies) [inline app/data/census_household.json:3632]: Should include be in bold too?

File: app/data/census_household.json [hunk 13]
```diff
@@ -3925,24 +4046,40 @@
                                             "mandatory": false,
                                             "options": [
                                                 {
-                                                    "label": "Yes",
-                                                    "value": "Yes"
+                                                    "label": "Yes, usually lives in the United Kingdom",
```
Comment (benjefferies) [inline app/data/census_household.json:4343]: These answers differ from the slides

File: app/templates/confirmation.html [hunk 14]
```diff
@@ -7,8 +7,8 @@
 <a class="u-dib" href="{{previous_location}}">Previous</a>
 
 <div class="header">
-  <h1 class="header__title saturn">Confirmation</h1>
-  <h2 class="header__subtitle">Thank you for your answers, do you wish to submit</h2>
+    <h1 class="header__title saturn">Confirmation</h1>		 +  <h1 class="header__title saturn">You’ve completed the 2017 Census Test</h1>
```
Comment (benjefferies) [inline app/templates/confirmation.html]: There looks to me some merging issues here

File: app/data/census_household.json (hunk 10 above)
Comment (jonnyshaw89) [inline app/data/census_household.json]: Im not sure thats supported? wasn't that on the original list?

File: app/templates/partials/questions/relationship.html [hunk 15]
```diff
@@ -2,7 +2,14 @@
...
 
-    <h2 class="neptune">{{question_title}}</h2> {{question_description|safe}}
+    <h2 class="question__title neptune">
+      {%- if question_number -%}
+        <span>{{question_number}}. </span>
+      {% endif %}
+      {{question_title|safe}}
+    </h2>
+    {{question_description|safe}}
+    {{question_guidance|safe}}
```
Comment (collisdigital) [inline app/templates/partials/questions/relationship.html:12]: Note, @hamishtaplin is going to look at presenting this differently (added a card: https://trello.com/c/Oc4yln5z)

**Review by iwootten — APPROVED** [review]

File: tests/functional/generate_pages.py [hunk 16]
```diff
@@ -52,6 +55,13 @@
 
 """
 
+DROP_DOWN_SETTER = r"""  set{answerName}(value) {
```
Comment (jonnyshaw89) [inline tests/functional/generate_pages.py:58]: I think these changes have been made in another branch by ben

Comment (collisdigital) [reply tests/functional/generate_pages.py:58]: Ah yes #660; this is enough to get this PR in, we can merge the two afterwards; Ben's also include the relationship question stuff

Comment (collisdigital) [reply tests/functional/generate_pages.py:58]: Andrew has had to make some similar changes too; the joys of parallel working 😀

File: app/data/census_household.json (hunk 10 above)
Comment (collisdigital) [reply app/data/census_household.json:3312]: Done

File: app/data/census_household.json (hunk 7 above)
Comment (collisdigital) [reply app/data/census_household.json:463]: Matches latest slides

File: app/data/census_household.json (hunk 4 above)
Comment (collisdigital) [reply app/data/census_household.json:1211]: Done

**Review by ajmaddaford — APPROVED** [review]

File: tests/functional/generate_pages.py [hunk 17]
```diff
@@ -31,7 +32,9 @@
 
 """
 
-HEADER = r"""import {basePage} from '../../{basePageFile}'
+HEADER = r"""// >>> WARNING THIS PAGE WAS AUTO-GENERATED ON {generated} - DO NOT EDIT!!! <<<
```
Comment (benjefferies) [inline tests/functional/generate_pages.py:35]: Should we be checking these files into the repository? They are compiled so fast it could be part of the build process
